go/master/etcd_client.go

```diff
@@ -173,6 +177,44 @@ func GetKey(c *clientv3.Client, key string, timeout time.Duration) (string, erro
 	return string(v), nil
 }
 
+// RegisterTrainer will generate a uuid and write as key to etcd for each trainer node.
+func RegisterTrainer(c *clientv3.Client, timeout time.Duration) error {
+	clientUUID, e := uuid.NewV4()
+	if e != nil {
+		return e
+	}
+	trainerIP, e := networkhelper.GetExternalIP()
+	if e != nil {
+		return e
+	}
+	// put trainer uniq id with etcd lease
+	resp, err := c.Grant(context.TODO(), 5)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	_, err = c.Put(ctx, DefaultTrainerPath+"/"+clientUUID.String(), trainerIP, clientv3.WithLease(resp.ID))
```

Comment:
How about write the pass status into etcd as the following?

```golang
type TrainerStatus struct {
    trainerIP string
    pass_num int
    pass_status string
}
```

For the master check a pass status:
1. master check trainer pass status:
    1.1. if all trainer finished a pass, goto 2
    1.2. else goto 1
2. master move task from Done Queue to Todo Queue.

For the trainer check a pass status:
1. fetch task from master
    1.1. if NoTaskFound, Update status to Finished and goto 2
    1.1. else training with the task
2. if pass_num == finised_pass_num; then finish the train; else goto 1.